src/gettext.h

```diff
@@ -77,3 +78,20 @@ inline std::wstring fwgettext(const char *src, Args&&... args)
 	delete[] str;
 	return std::wstring(buf);
 }
+
+template <typename ...Args>
+inline std::string fmtgettext(const char *format, std::size_t buf_size, Args&&... args)
+{
+	format = gettext(format);
+	if (buf_size <= 0) {
+		buf_size = porting::mt_snprintf(nullptr, 0, format, std::forward<Args>(args)...) + 1;
+		if (buf_size <= 0) {
+			throw std::runtime_error("gettext format error: " + std::string(format));
+		}
+	}
+	auto size = static_cast<size_t>(buf_size);
+	char buf[size];
```

Comment:
[VLA](https://en.wikipedia.org/wiki/Variable-length_array)s are not portable, you can use a string directly like this:
```cpp
std::string buf;
buf.resize(size+1); // extra null byte
porting::mt_snprintf(&buf[0], buf.size(), /* ... */);
buf.resize(size);
return buf;
```

Reply:
ok.